lti_consumer/lti_xblock.py

```diff
@@ -691,6 +691,13 @@ def validate_field_data(self, validation, data):
                 _('Custom Parameters should be strings in "x=y" format.'),
             )))
 
+        # Validate external config ID is not missing.
+        if data.config_type == 'external' and not data.external_config:
+            _ = self.runtime.service(self, 'i18n').ugettext
+            validation.add(ValidationMessage(ValidationMessage.ERROR, str(
+                _('Reusable configuration ID must be set when using external config.'),
+            )))
```

Comment:
I added a validation to the external ID format on the latest commit.

Reply:
The regex check will raise TypeError if data.external_config is None, but since the value is being transformed into str that will not happen, and the regex validation will fail to validate a string with a value of 'None'. I just applied your suggestion to mix both validations in a single condition and also to improve the validation on the model.